optuna/trial/_trial.py

```diff
@@ -485,6 +485,14 @@ def objective(trial):
             )
             raise TypeError(message) from None
 
+        try:
+            step = int(step)
+        except (TypeError, ValueError):
```

Comment:
Is it noted anywhere in the documentation that these are the only two exceptions that can occur with a cast to int?

Reply:
Thanks! TBH, I just copied the code block for `value`'s cast and  I've just found https://stackoverflow.com/a/74752705/2406562.

Reply:
Thank you. I think that any exception can be thrown from `__int__`, but `TypeError` and `ValueError` seem reasonable.